configure.ac

```diff
@@ -970,6 +970,8 @@ if test "$TARGET_OS" = "darwin"; then
   AX_CHECK_LINK_FLAG([-Wl,-bind_at_load], [HARDENED_LDFLAGS="$HARDENED_LDFLAGS -Wl,-bind_at_load"], [], [$LDFLAG_WERROR])
 fi
 
+AX_CHECK_LINK_FLAG([-Wl,--build-id=0xfedcba9876543210], [LDFLAGS="$LDFLAGS -Wl,--build-id=0xfedcba9876543210"], [AC_MSG_ERROR([linker did not accept --build-id flag])], [$LDFLAG_WERROR])
```

Comment:
Why is this an error if it's not accepted? We can't use it that way, because not all of our linkers support this option.

Reply:
> Why is this an error if it's not accepted?

Mostly, to ensure GUIX build are successful for all hosts.

> We can't use it that way, because not all of our linkers support this option.

Agree. Will be happy if a more suitable approach will be suggested.

Reply:
Not opposed to specifying this, but perhaps we'd do it at the `contrib/guix/libexec/build.sh` level as a `HOST_LDFLAGS`